protocol/src/lib.rs

```diff
@@ -132,32 +132,68 @@ pub struct SnapshotDetails {
     pub snapshot_name: String,
 }
 
+/**
+ * Crucible Upstairs Downstairs message protocol version.
+ * Crude, but its something.  This should be changed whenever there is
+ * any change to the enum Message below.
+ */
+pub const CRUCIBLE_MESSAGE_VERSION: u32 = 2;
```

Comment:
A pattern we have in propolis for managing the (crude) versioning with bhyve uses a u32 like this to represent the current version, but its type is an `IntoPrimitive` enum: https://github.com/oxidecomputer/propolis/blob/master/crates/bhyve-api/src/lib.rs#L256-L287. I like this approach because it allows us to embed in the code some notes about what changed during versions. It might be worth doing something like this here, since I imagine crucible protocol changes might be even more gnarly than bhyve ones, and it could be worthwhile to have notes like this.

Reply:
See if the new changes are what you were thinking of.

Reply:
No problem, definitely don't want to block on that idea.